source/game/system/map/MapdataCheckPoint.cc

```diff
@@ -0,0 +1,258 @@
+    m_dir = EGG::Vector2f(left().y - right().y, right().x - left().x);
+    m_dir.normalise();
+}
+
+void MapdataCheckPoint::read(EGG::Stream &stream) {
+    m_left.read(stream);
+    m_right.read(stream);
+    m_jugemIndex = stream.read_u8();
+    m_type = stream.read_s8();
+    m_prevPt = stream.read_u8();
+    m_nextPt = stream.read_u8();
+}
+
+/// @addr{0x80510D7C}
+/// @see MapdataCheckPoint::checkSectorAndCheckpointCompletion_
+MapdataCheckPoint::SectorOccupancy MapdataCheckPoint::checkSectorAndCheckpointCompletion(
+        const EGG::Vector3f &pos, float *completion) const {
+    EGG::Vector2f p1 = right();
+    p1.y = pos.z - p1.y;
+    p1.x = pos.x - p1.x;
+
+    for (s32 nextIdx = 0; nextIdx < m_nextCount; nextIdx++) {
+        EGG::Vector2f p0 = m_nextPoints[nextIdx].checkpoint->left();
+        p0.y = pos.z - p0.y;
+        p0.x = pos.x - p0.x;
+        MapdataCheckPoint::SectorOccupancy result =
+                checkSectorAndCheckpointCompletion_(m_nextPoints[nextIdx], p0, p1, completion);
+
+        if (result == OutsideSector) {
+            continue;
+        } else {
+            return result;
+        }
+    }
+
+    return OutsideSector;
+}
+
+bool MapdataCheckPoint::isPlayerFlagged(s32 /* playerIdx */) const {
+    return m_flag;
+}
+
+void MapdataCheckPoint::setPlayerFlags(s32 /* playerIdx */) {
+    m_flag = true;
+}
+
+void MapdataCheckPoint::resetFlags() {
+    m_flag = false;
+}
+
+EGG::Vector2f MapdataCheckPoint::left() const {
+    return m_left;
+}
+
+EGG::Vector2f MapdataCheckPoint::right() const {
+    return m_right;
+}
+
+u8 MapdataCheckPoint::jugemIndex() const {
+    return m_jugemIndex;
+}
+
+s8 MapdataCheckPoint::type() const {
+    return m_type;
+}
+
+bool MapdataCheckPoint::isNormalCheckpoint() const {
+    return type() == NORMAL_CHECKPOINT;
+}
+
+bool MapdataCheckPoint::isFinishLine() const {
+    return type() == FINISH_LINE;
+}
+
+u8 MapdataCheckPoint::prevPt() const {
+    return m_prevPt;
+}
+
+u8 MapdataCheckPoint::nextPt() const {
+    return m_nextPt;
+}
+
+u16 MapdataCheckPoint::nextCount() const {
+    return m_nextCount;
+}
+
+u16 MapdataCheckPoint::prevCount() const {
+    return m_prevCount;
+}
+
+u16 MapdataCheckPoint::id() const {
+    return m_id;
+}
+
+MapdataCheckPoint *MapdataCheckPoint::prevPoint(s32 i) const {
+    return m_prevPoints[i];
+}
+
+MapdataCheckPoint *MapdataCheckPoint::nextPoint(s32 i) const {
+    return m_nextPoints[i].checkpoint;
+}
+
+const LinkedCheckpoint &MapdataCheckPoint::nextLinked(s32 i) const {
+    return m_nextPoints[i];
+}
+
+/// @brief computes @ref m_prevKcpId for each checkpoint.
+/// @details @ref isPlayerFlagged indicates that a checkpoint has been visited, to prevent infinite
+/// recursion.
+/// https://decomp.me/scratch/MtF18
+/// @addr{0x80515A6C}
+void MapdataCheckPoint::linkPrevKcpIds(u8 prevKcpId) {
+    m_prevKcpId = isNormalCheckpoint() ? prevKcpId : this->type();
```

Comment:
```suggestion
    m_prevKcpId = isNormalCheckpoint() ? prevKcpId : m_type;
```

Reply:
`m_type` or `type()`? i use `type()` accross this file. should i switch them all to `m_type` or write this as `m_prevKcpId = isNormalCheckpoint() ? prevKcpId : type()`?

Reply:
Yeah I would just switch them all to `m_type`